contracts/cw721-base/src/state.rs

```diff
@@ -1,152 +1,61 @@
-use schemars::JsonSchema;
-use serde::de::DeserializeOwned;
-use serde::{Deserialize, Serialize};
-use std::marker::PhantomData;
-
-use cosmwasm_std::{Addr, BlockInfo, CustomMsg, StdResult, Storage};
-
-use cw721::{ContractInfoResponse, Cw721, Expiration};
-use cw_storage_plus::{Index, IndexList, IndexedMap, Item, Map, MultiIndex};
-
-pub struct Cw721Contract<'a, T, C, E, Q>
-where
-    T: Serialize + DeserializeOwned + Clone,
-    Q: CustomMsg,
-    E: CustomMsg,
+// expose to all others using contract, so others dont need to import cw721
+pub use cw721::state::*;
+use cw721::traits::{Cw721CustomMsg, Cw721State};
+
+#[deprecated(since = "0.19.0", note = "Please use `NftInfo`")]
+pub type TokenInfo<TNftMetadataExtension> = NftInfo<TNftMetadataExtension>;
+
+pub struct Cw721Contract<
+    'a,
+    // Metadata defined in NftInfo (used for mint).
+    TNftMetadataExtension,
+    // Message passed for updating metadata.
+    TNftMetadataExtensionMsg,
+    // Extension defined in CollectionMetadata.
+    TCollectionMetadataExtension,
+    TCollectionMetadataExtensionMsg,
```

Comment:
```suggestion
    // Extension defined in CollectionMetadata.
    TCollectionMetadataExtension,
    // Message passed for updating collection metadata.
    TCollectionMetadataExtensionMsg,
```

Comments are good, might as well add this one.